### ISSUE TITLE
Encoding handling broken when doing <anysource>-to-scalar[array] tiyding

### ISSUE THREAD

**Issue by kenneth-olwing** [perltidy/perltidy#83]
Hi,

I'm using Code::TidyAll, and recently we've been converting our codebase to UTF-8, and using '-utf8' to perltidy. However, I noticed that some files became tidied despite that they should already *be* tidied. After tidying, there was no code change - except that the file had been reverted back to ISO-8859-1. Investigating this I realized that the files being tidied were source files with comments/podtext in swedish and using some national characters ('åäö'). However, I couldn't get this to happen from the command line using perltidy - then it worked fine.

After tracing things in Code::TidyAll I realized that C::T calls directly into Perl::Tidy::perltidy() with source and destination as scalar references. Debugging at that point, it was clear that the source has the UTF-8 encoded source, but the destination comes back as an unencoded string. Hence, P::T doesn't honor the encoding correctly in this case (the same problem occurs when using scalar arrays).

I've tracked the problem down and a PR on this is forthcoming shortly. I've managed a workaround in my C::T usage (a custom plugin), but it'd be great to have it fixed in the official code.

For a demonstration of what's happening, I'm attaching a test script that reads a test file with encoded strings only; not really runnable and shouldn't be tidied in any way and so is expected to come back with the exact same data as went in. Using the latest code from this repository, the script becomes unhappy with this; the PR will make it work.

Thanks for P::T, apart from this issue we have had great use of it for a long time and will continue to use it heavily.

ken1

[tidyperltests.zip](https://github.com/perltidy/perltidy/files/7892507/tidyperltests.zip)

**Comment by shancock9** [comment]
Thanks for finding and working on this Ken, much appreciated. I'll start working on it.

**Comment by shancock9** [comment]
Version 20220217 is out with the new -eos flag.    

Ken, thanks again for working on this!  I posted a note at the https://github.com/houseabsolute/perl-code-tidyall/issues/84 to be sure tidyall users were aware of it.  And Dave Rolsky has already updated Code::TidyAll to use the flag, so this problem will go away when users update Code::TidyAll and Perl::Tidy.

I encountered a little unexpected trouble with the new test cases (they were reading a local `.perltidyrc`!) but it was easy to figure out and fix.

I plan to change the default in Perl::Tidy to be `-eos` in a future release, since it is clearly the most useful default.
